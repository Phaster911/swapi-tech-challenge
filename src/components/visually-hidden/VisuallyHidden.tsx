// These styles will make sure the component
// is not visible, but will still be announced
// by screen readers.

import { CSSProperties, ReactNode } from "react";

const hiddenStyles: CSSProperties = {
  display: "inline-block",
  position: "absolute",
  overflow: "hidden",
  clip: "rect(0 0 0 0)",
  height: 1,
  width: 1,
  margin: -1,
  padding: 0,
  border: 0,
};

type Props = {
  children: ReactNode;
};

export default function VisuallyHidden({ children }: Props) {
  return <span style={hiddenStyles}>{children}</span>;
}
