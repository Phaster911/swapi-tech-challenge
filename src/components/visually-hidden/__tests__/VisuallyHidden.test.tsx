import { render } from "@testing-library/react";
import VisuallyHidden from "../VisuallyHidden";

describe("<VisuallyHidden />", () => {
  it("renders the component", () => {
    const { container } = render(
      <VisuallyHidden>
        <span></span>
      </VisuallyHidden>
    );
    expect(container).toMatchSnapshot();
  });
});
