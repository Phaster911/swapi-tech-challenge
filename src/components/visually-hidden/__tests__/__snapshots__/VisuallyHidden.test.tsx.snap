// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<VisuallyHidden /> > renders the component 1`] = `
<div>
  <span
    style="display: inline-block; position: absolute; overflow: hidden; height: 1px; width: 1px; margin: -1px; padding: 0px; border: 0px;"
  >
    <span />
  </span>
</div>
`;
